### PR TITLE
docs:#817 fix docker quickstart

### LINKED ISSUE #817: Docker quickstart examples don't work
The examples in the following link don't work. For one thing, it seems that a line was added without a proper line continuation character `\` which causes the command to be broken up:

```
--overprovisioned \
--set redpanda.empty_seed_starts_cluster=false
--seeds "redpanda-1:33145" \
```

https://docs.redpanda.com/docs/platform/quickstart/quick-start-docker/

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #817 

This fixes the examples. It also fixes a link and makes the headings consistent in the Docker and Windows installs. 

preview: 
- https://deploy-preview-820--redpanda-documentation.netlify.app/docs/platform/quickstart/quick-start-docker/
- https://deploy-preview-820--redpanda-documentation.netlify.app/docs/platform/quickstart/quick-start-windows/

### CODE REVIEW

**Review by andrwng — APPROVED** [review]

File: docs/platform/quickstart/quick-start-docker.mdx [hunk 1]
```diff
@@ -104,7 +103,7 @@ docker.redpanda.com/vectorized/redpanda redpanda start \
 --memory 1G  \
 --reserve-memory 0M \
 --overprovisioned \
---set redpanda.empty_seed_starts_cluster=false
+--set redpanda.empty_seed_starts_cluster=false \
```
Comment (dlex) [inline docs/platform/quickstart/quick-start-docker.mdx:106]: The 3rd `docker run` is missing this line

Comment (micheleRP) [reply docs/platform/quickstart/quick-start-docker.mdx:106]: Thanks, @dlex! I've updated the [preview draft](https://deploy-preview-820--redpanda-documentation.netlify.app/docs/platform/quickstart/quick-start-docker/#set-up-a-three-node-cluster). One additional question: Esteban suggested adding --mode dev-container (see https://redpandadata.slack.com/archives/C01LK3UK2K1/p1667950545249809). Should we add that to the Docker and Windows quickstarts in this PR?

Comment (dlex) [reply docs/platform/quickstart/quick-start-docker.mdx:106]: I would say let's deal with that separately, that doesn't seem to be related to #817.

**Review by dlex — APPROVED** [review]

**Review by Deflaimun — APPROVED** [review]

File: docs/platform/quickstart/quick-start-windows.mdx [hunk 2]
```diff
@@ -13,7 +13,9 @@ For production or benchmarking, see [Deploying for Production](../../deployment/
 
 ## Prerequisites
 
-Before you start, you must install [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install) and [Docker for Windows](https://docs.docker.com/desktop/windows/install/). Additionally, for Docker for Windows to work, you must [enable your Hypervisor service](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v) in the Windows Control Panel.
+- Make sure you have [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install) installed.
+- Make sure you have [Docker for Windows](https://docs.docker.com/desktop/windows/install/) installed. 
+- Enable your [Hypervisor service](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v) in the Windows Control Panel (required for Docker for Windows).
```
Comment (AnneLeeperRP) [inline docs/platform/quickstart/quick-start-windows.mdx:18]: What do you think about adding a lead-in sentence to introduce the bullet list? For example, the doc for "Installing Redpanda on Kubernetes" says:
"Verify that you have the following software installed:"

**Review by AnneLeeperRP — APPROVED** [review]

File: docs/platform/quickstart/quick-start-windows.mdx (hunk 2 above)
Comment (micheleRP) [reply docs/platform/quickstart/quick-start-windows.mdx:18]: Yes, @anne, that's better!

**Review by AnneLeeperRP — APPROVED** [review]

**Review by dlex — APPROVED** [review]

**Comment by micheleRP** [comment]
Confirmed by Patrick Angeles.

**Comment by patrickangeles** [comment]
I just checked this again on a clean directory... the Docker compose example doesn't work.

```
INFO  2022-11-22 21:52:42,818 [shard 0] main - application.cc:1428 - Started RPC server listening at {host: 0.0.0.0, port: 33145}
INFO  2022-11-22 21:52:42,819 [shard 0] cluster - cluster_discovery.cc:147 - Requesting node ID for node UUID {33ff4fa6-572d-467f-82c6-004b6b45c22e} from {host: redpanda-1, port: 33145}
INFO  2022-11-22 21:52:49,152 [shard 0] cluster - cluster_discovery.cc:147 - Requesting node ID for node UUID {33ff4fa6-572d-467f-82c6-004b6b45c22e} from {host: redpanda-1, port: 33145}
INFO  2022-11-22 21:52:54,660 [shard 0] cluster - cluster_discovery.cc:147 - Requesting node ID for node UUID {33ff4fa6-572d-467f-82c6-004b6b45c22e} from {host: redpanda-1, port: 33145}
INFO  2022-11-22 21:53:00,307 [shard 0] cluster - cluster_discovery.cc:147 - Requesting node ID for node UUID {33ff4fa6-572d-467f-82c6-004b6b45c22e} from {host: redpanda-1, port: 33145}
INFO  2022-11-22 21:53:07,106 [shard 0] cluster - cluster_discovery.cc:147 - Requesting node ID for node UUID {33ff4fa6-572d-467f-82c6-004b6b45c22e} from {host: redpanda-1, port: 33145}
```

**Comment by andrwng** [comment]
This is probably the same issue that Josh posted in this thread: https://redpandadata.slack.com/archives/C01ND4SVB6Z/p1669142643154109

@patrickangeles can you try with the following line added to the compose file:

```
--advertise-rpc-addr redpanda-1:33145
```

**Comment by micheleRP** [comment]
@andrwng does it also require removing `--smp 1`? See https://redpandacommunity.slack.com/archives/C01AJDUT88N/p1668929030743389

**Comment by andrwng** [comment]
> @andrwng does it also require removing `--smp 1`? See https://redpandacommunity.slack.com/archives/C01AJDUT88N/p1668929030743389

No, `--smp 1` shouldn't be related to the issue here.

**Comment by andrwng** [comment]
I confirmed that adding this to the compose file results in a properly formed single-node cluster:

```
    ...
    - --advertise-pandaproxy-addr
    - PLAINTEXT://redpanda:28082,OUTSIDE://localhost:8082
    - --advertise-rpc-addr redpanda-1:33145
```

@micheleRP can you make this change?

**Comment by micheleRP** [comment]
> ...
>     - --advertise-pandaproxy-addr
>     - PLAINTEXT://redpanda:28082,OUTSIDE://localhost:8082
>     - --advertise-rpc-addr redpanda-1:33145

@andrwng yes, please confirm that I should add that line to both docker-compose and one-node set up for both Docker and Windows!

**Comment by andrwng** [comment]
@micheleRP good call, yes please add the line to every Docker deployment that doesn't already have it (I think that is just the Docker and Windows pages)

**Pull request merged**